### PR TITLE
VAR-424, VAR-430 | Add accessibility details to resources

### PR DESCRIPTION
**Pull request by ahie**
## Description :sparkles:

Implements fetching of accessibility data for resources and displays the required details in a separate panel on the resource page.

### Closes :no_good_woman:

[VAR-430](https://helsinkisolutionoffice.atlassian.net/browse/VAR-430)

### Automated tests :gear:️

-  Created `src/common/api/__test__/accessibilityClient.test.js`
- Added tests to `app/pages/resource/resource-info/__tests__/ResourceInfo.test.js`

### Manual testing :construction_worker_man:

- Open a resources page
- Accessibility information should be shown, if available through the API
  - An example resource with available accessibility data on the test server: Ryhmätila (itäkeskuksen kirjasto)
  - If no accessibility data is received, the panel should not be rendered

## Screenshots :camera_flash:

kuva

## Additional notes :spiral_notepad:

- The API calls should take into account the current users profile settings. Currently the page reports _all_ known accessibility shortcomings.
- Need to configure two additional environment variables, added these to `.env.example`:
  - ACCESSIBILITY_API_URL
  - ACCESSIBILITY_API_SYSTEM_ID

### CODE REVIEW

File: app/pages/resource/resource-info/ResourceInfo.js [hunk 1]
```diff
@@ -71,12 +94,94 @@ function ResourceInfo({
...
+              <div>
+                <h4>{t('ResourceInfo.accessibilityShortcomingsTitle')}</h4>
+                {accessibilityInformation.shortcomings.map(
+                  (sentenceGroup, sentenceGroupId) => (
+                    <React.Fragment
+                      key={`accessibility-sentence-group-${sentenceGroupId}`}
+                    >
+                      <div className="app-ResourceInfo__accessibility__shortcoming-sentence-group">
+                        <img
+                          alt="accessibility-shortcoming"
```
Comment (Nizar-Rahme) [inline app/pages/resource/resource-info/ResourceInfo.js:164]: Should we add this to locale files?

File: app/pages/resource/resource-info/ResourceInfo.js [hunk 2]
```diff
@@ -71,12 +94,94 @@ function ResourceInfo({
...
+                          className="app-ResourceInfo__accessibility__icon"
+                          src={iconAccessibilityError}
+                        />
+                        <b>{sentenceGroup.sentenceGroup}</b>
```
Comment (Nizar-Rahme) [inline app/pages/resource/resource-info/ResourceInfo.js:168]: `strong` semantically emphasizes the respective text as important. 
```suggestion
                        <strong>{sentenceGroup.sentenceGroup}</strong>
```

File: src/common/api/accessibilityClient.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  getShortcomings(resourceId, locale = 'fi') {
+    // eslint-disable-next-line
+    return Promise.all([this.apiClient.get(`targets/${this.systemId}/${resourceId}/shortages`), this.apiClient.get('viewpoints')])
+      .then(([shortagesRes, viewpointsRes]) => {
+        const idToNameMap = {};
+
+        viewpointsRes.data.forEach((viewpoint) => {
+          idToNameMap[viewpoint.viewpointId] = viewpoint.names.find(name => name.language === locale).value;
```
Comment (Nizar-Rahme) [inline src/common/api/accessibilityClient.js:65]: In some edge cases, `find` return value might be `undefined`.

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
Overall it looks good to me, left a couple of comments.

File: src/common/api/accessibilityClient.js (hunk 3 above)
Comment (ahie) [reply src/common/api/accessibilityClient.js]: Added a function to resolve fallback locales (defaults to fi).

File: app/pages/resource/resource-info/ResourceInfo.js (hunk 2 above)
Comment (ahie) [reply app/pages/resource/resource-info/ResourceInfo.js]: Done.

File: app/pages/resource/resource-info/ResourceInfo.js (hunk 1 above)
Comment (ahie) [reply app/pages/resource/resource-info/ResourceInfo.js]: Done.

File: app/i18n/messages/fi.json [hunk 4]
```diff
@@ -335,6 +335,11 @@
   "ResourceInfo.loginMessage": "Sinun täytyy <a href=\"/login\">kirjautua sisään</a>, jotta voit tehdä varauksen tähän tilaan.",
   "ResourceInfo.reservationTitle": "Varauksen tiedot",
   "ResourceInfo.reserveTitle": "Varaa",
+  "ResourceInfo.accessibilityTitle": "Esteettömyys",
+  "ResourceInfo.accessibilityDetailsTitle": "Resurssin esteettömyystiedot",
+  "ResourceInfo.accessibilityShortcomingsTitle": "Resurssin esteettömyyspuutteet",
+  "ResourceInfo.noAccessibilityShortcomings": "Ei tunnettuja esteettömyyspuutteita",
+  "ResourceInfo.foundAccessibilityShortcomings": "{nShortcomings} esteettömyyspuutetta profiilillesi",
```
[the remark this answers is not in the document]
Comment (ahie) [reply app/i18n/messages/fi.json:342]: Actually, single curly braces are correct here. Changed to those in all translations.

**Review by Nizar-Rahme — APPROVED** [review]
Rebasing this branch and I think it's ready for :shipit:

**Comment by ahie** [comment]
Rebased.

**Pull request merged**